file_zbc.c

```diff
@@ -2027,6 +2032,86 @@ static int zbc_read(struct tcmu_device *dev, struct tcmulib_cmd *cmd)
 	return SAM_STAT_GOOD;
 }
 
+/*
+ * Check conv to seq and seq to seq zone boundary crossings.
+ * and write pointer alignment in sequential write required zones.
+ */
+static int zbc_write_check_zones(struct tcmu_device *dev,
+				 struct tcmulib_cmd *cmd,
+				 size_t nr_lbas, uint64_t lba)
+{
+	struct zbc_dev *zdev = tcmu_get_dev_private(dev);
+	struct zbc_zone *zone;
+	int zone_type = 0;
+	size_t count;
+
+	while (nr_lbas) {
+
+		/* Get the zone of the current LBA */
+		zone = zbc_get_zone(zdev, lba, false);
+		if (!zone) {
+			tcmu_dev_err(zdev->dev,
+				     "Get zone at LBA %llu failed\n",
+				     lba);
+			return tcmu_set_sense_data(cmd->sense_buf,
+						   HARDWARE_ERROR,
+						   ASC_INTERNAL_TARGET_FAILURE,
+						   NULL);
+		}
+
+		/* Check conv -> seq and seq -> seq zone boundary crossing */
+		if (zone_type == 0)
+			zone_type = zone->type;
+		if (zone->type != zone_type ||
+		    (zbc_zone_seq_req(zone) &&
+		     lba + nr_lbas > zone->start + zone->len)) {
+			tcmu_dev_err(dev,
+				     "Write boundary violation lba %llu, "
+				     "xfer len %lu\n",
+				     lba, nr_lbas);
```

Comment:
Just a minor nit. I am not strict on coding style as long as it does not get too different form the core code where we try to use the kernel coding style because we have to go back and forth between the target_core_user module and user space.

However, it seems in this patch we are adding a new style. I think normally in file_zbc we are not breaking up the string even if it goes over 80 chars, so we would do

tcmu_dev_err(dev, "Write boundary violation lba %llu, "xfer len %lu\n",

but here and some places below you add the new style that strictly follows 80 chars and breaks it up.

Reply:
Sure, let's keep it consistent with the rest of the codebase. I have amended the code not to break up lines. Also ran checkpatch and fixed a few things it complained about.